client/carwash.lua

```diff
@@ -42,21 +48,22 @@ CreateThread(function()
         local dirtLevel = GetVehicleDirtLevel(cache.vehicle)
         sleep = 1000
         if IsPedInAnyVehicle(cache.ped, false) then
-            for k in pairs(Config.CarWash) do
-                local dist = #(PlayerPos - vector3(Config.CarWash[k]['coords']['x'], Config.CarWash[k]['coords']['y'], Config.CarWash[k]['coords']['z']))
+            for i = 1, #Config.CarWash.locations do
+                local carWashCoords = Config.CarWash.locations[i].coords
+                local dist = #(PlayerPos - vector3(carWashCoords.x, carWashCoords.y, carWashCoords.z))
                 if dist <= 7.5 and Driver then
                     sleep = 0
                     if not washingVehicle then
-                        DrawText3Ds(Config.CarWash[k]['coords']['x'], Config.CarWash[k]['coords']['y'], Config.CarWash[k]['coords']['z'], '~g~E~w~ - Washing car ($'..Config.DefaultPrice..')')
+                        DrawText3Ds(carWashCoords.x, carWashCoords.y, carWashCoords.z, '~g~E~w~ - Wash the car ($'..Config.CarWash.defaultPrice..')')
                         if IsControlJustPressed(0, 38) then
-                            if dirtLevel > Config.DirtLevel then
+                            if dirtLevel > Config.CarWash.dirtLevel then
                                 TriggerServerEvent('qb-carwash:server:washCar')
                             else
                                 QBCore.Functions.Notify('The vehicle isn\'t dirty', 'error')
                             end
                         end
                     else
-                        DrawText3Ds(Config.CarWash[k]['coords']['x'], Config.CarWash[k]['coords']['y'], Config.CarWash[k]['coords']['z'], 'The car wash is not available ..')
+                        DrawText3Ds(carWashCoords.x, carWashCoords.y, carWashCoords.z, 'The car wash is not available...')
```

Comment:
Unrelated; we should use the import Draw3DText and remove this local function

Reply:
How would you like this done? So I can do it properly in the future?

Reply:
Might go do all that here shortly